sql/connect/server/src/main/scala/org/apache/spark/sql/connect/pipelines/PipelinesHandler.scala

```diff
@@ -156,6 +156,10 @@ private[connect] object PipelinesHandler extends Logging {
               .filter(_.nonEmpty),
             properties = dataset.getTablePropertiesMap.asScala.toMap,
             baseOrigin = QueryOrigin(
+              filePath = Option.when(dataset.getSourceCodeLocation.hasFileName)(
```

Comment:
nit: we can store `filePath` and `line` to variables or a method to avoid duplicated code